provider/zonefinder.go

```diff
@@ -25,8 +29,12 @@ func (z ZoneIDName) Add(zoneID, zoneName string) {
 }
 
 func (z ZoneIDName) FindZone(hostname string) (suitableZoneID, suitableZoneName string) {
+	name, err := idna.Lookup.ToUnicode(hostname)
+	if err != nil {
+		name = hostname
+	}
```

Comment:
Is this really enough ? What about a warning log here to (at least) inform user of the error ?

Reply:
Sure, I'll add something. Is this okay?
```go
log.Warnf("Failed to normalise '%s': %v", hostname, err)
```

Reply:
Could we add test as well for that logging entry? https://github.com/kubernetes-sigs/external-dns/blob/4b2fd495d944f9a67cdeb5260db95c5f2367d8bf/internal/testutils/log.go#L36. you could find example usages in code

Reply:
```go
log.Warnf("Failed to convert hostname '%s' to its Unicode form: %v", hostname, err)
```

Reply:
Looks good, I'll get to adding the warning and the respective test later today. 👍🏻